File: code/matMul.cpp
#include <stdlib.h>

int ** randomMatrix(int n){
    int** r;
    r = new int*[n];
    int i,j;
    for (i = 0; i < n; ++i) {
        for (j = 0;  j< n; ++j){
            r[i][j] = rand()%100;
        }
    }
    return r;
}

int ** matMul(int n ,int** m1,int** m2){
    int **r;
    r = new int*[n];
    int i,j,k;
    for (i = 0; i < n; ++i) {
        for (j = 0; j< n; ++j ) {
            int sum = 0;
            for (k = 0; k < n; ++k) {
                sum = sum + m1[i][k]* m2[k][j]; 
            }
            r[i][j] = sum;
        }
    }
    return r;
}
int main(void){
    int n = 1000;
    int ** r = randomMatrix(n);
    int ** h = matMul(n,r,r);
    return 0;
}

File: code/matMul.js
let n = 600
//Function to generate random Matrix
function randomMatrix(n) {
    let a = []
    for (let i = 0; i < n ; i++) {
        let aux = []
        for (let j = 0; j < n ; j++) {
            aux.push(Math.floor(Math.random() * 10));   
        }
        a.push(aux)
    }
    return a;
}

//Function to multiplcate
function matMul(a,b,n) {
    let res = [];
    for (let i = 0, len = n; i < len; i++) {
        let aux = [];
        for (let j = 0, len = n; j < len; j++) {
            let sum = 0;
            for (let k = 0, len = n; k < len; k++) {
                sum += a[i][k]*b[k][j];
            } 
            aux.push(sum);
        }
        res.push(aux);
    }
    return res;
}

let r = randomMatrix(n)
let s = matMul(r,r,n)


